[FW][ADD] account_payment_group_financial_surcharge: compatibility with AR EDI

fw-bot-adhoc opened the pull request:
Add conext key in order to be used to identify and auto complete missing fields required for EDI

Forward-Port-Of: ingadhoc/account-payment#349

roboadhoc commented:
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/account-payment/pull/350).

fw-bot-adhoc commented:
@zaoral @ica-adhoc cherrypicking of pull request ingadhoc/account-payment#349 failed.

* 2181083a6de4ea98455c4019ee25ba70a1c8b4d8
* 31ff78fec4186efd75833b3fc5fe3b6480c77043 <- on this commit

stderr:
```
20:34:09.374824 git.c:444               trace: built-in: git cherry-pick 31ff78fec4186efd75833b3fc5fe3b6480c77043
error: Cherry-picking is not possible because you have unmerged files.
hint: Fix them up in the work tree, and then use 'git add/rm <file>'
hint: as appropriate to mark resolution and make a commit.
fatal: cherry-pick failed
----------
status:

```

Either perform the forward-port manually (and push to this branch, proceeding as usual) or close this PR (maybe?).

In the former case, you may want to edit this PR message as well.

zaoral committed:
[ADD] account_payment_group_financial_surcharge: Avoid ND = 0

If the financial plan does not add any subcharge we should not
create a ND amount 0. Only create the ND when we have a subcharge to
apply

Original commit: 546415cd8111d7e3a9e257a517504f909903979a

zaoral committed:
[ADD] account_payment_group_financial_surcharge: compatibility with AR edi

Add conext key in order to be used to identify and auto complete missing
fields required for EDI

Original commit d334b536b5b98034fb7018450bdbdbc460ea1e4f

zaoral commented:
@roboadhoc r+
